src/compiler/commandLineParser.ts

```diff
@@ -3233,7 +3245,7 @@ namespace ts {
     }
 
     function convertJsonOptionOfListType(option: CommandLineOptionOfListType, values: readonly any[], basePath: string, errors: Push<Diagnostic>): any[] {
-        return filter(map(values, v => convertJsonOption(option.element, v, basePath, errors)), v => !!v);
+        return filter(map(values, v => convertJsonOption(option.element, v, basePath, errors)), v => option.listPreserveFalsyValues ? true : !!v);

```

Comment:
i think instead of this we should just check if option is defined and get rid of `listPreserveFalsyValues`

Reply:
That would change the behavior of other lists, like `types`, which could impact existing projects. Why risk exposing them?

To be clear what you're looking for, the filter condition would become `v => v !== undefined`?

Reply:
@sheetalkamat Would you like to make this change, or is the risk of impacting existing projects too great? Please advise.